app/models/operatingsystem.rb

```diff
@@ -170,7 +170,14 @@ def self.find_by_to_label(str)
     os = find_by_description(str.to_s)
     return os if os
     a = str.split(" ")
-    b = a[1].split('.') if a[1]
+    if a[0] == 'Ubuntu'
```

Comment:
This is getting hard to read, so I started to rewrite it:
```ruby
name, version = str.split(" ")
major = minor = nil
if name == 'Ubuntu'
  if version
    x, y, minor = version.split('.', 3)
    major = "#{x}.#{y}"
  end
else
  major, minor = version.split('.') if version
end

cond = {:name => name}
cond[:major] = major if major
cond[:minor] = minor if minor

find_by(cond)
```

Then it made me question the logic. So how about:
```ruby
name, version = str.split(" ")

cond = {:name => name}

if version
  if name == 'Ubuntu'
    x, y, minor = version.split('.', 3)
    major = "#{x}.#{y}"
  else
    major, minor = version.split('.')
  end

  cond[:major] = major if major
  cond[:minor] = minor if minor
end

find_by(cond)
```

But this function needs some tests to cover this.

Reply:
Thanks, @ekohl, I've changed the method as suggested and added tests for Debian and Ubuntu. In the course of this I added Debian 12 to the OS fixtures and updated the Ubuntu 10.10 OS. Or is there a specific reason this old version is used?